ddtrace/tracer/span.go

```diff
@@ -481,7 +481,12 @@ func (s *span) Finish(opts ...ddtrace.FinishOption) {
 	}
 
 	if tr, ok := internal.GetGlobalTracer().(*tracer); ok && tr.rulesSampling.traces.enabled() {
-		tr.rulesSampling.SampleTrace(s)
+		s.context.trace.mu.RLock()
+		locked := s.context.trace.locked
+		s.context.trace.mu.RUnlock()
```

Comment:
We shouldn't be locking a trace's internal mutex from a span. Let's wrap this in a safe method.

Reply:
Added the helper methods

Reply:
@dianashevchenko Could we use the function `isTraceLocked` instead?

```suggestion
		locked := s.context.trace.isTraceLocked()
```